mne/preprocessing/faster.py

```diff
@@ -0,0 +1,368 @@
+        crossing this threshold value is marked as bad. Defaults to 3.
+    use_metrics : list of str
+        List of metrics to use. Can be any combination of:
+            'variance', 'correlation', 'hurst', 'kurtosis', 'line_noise'
+        Defaults to all of them.
+
+    Returns
+    -------
+    bads : list of str
+        The names of the bad EEG channels.
+    """
+    metrics = {
+        'variance':    lambda x: np.var(x, axis=1),
+        'correlation': lambda x: nanmean(
+                           np.ma.masked_array(
+                               np.corrcoef(x),
+                               np.identity(len(x), dtype=bool)
+                           ),
+                           axis=0),
+        'hurst':       lambda x: _hurst(x),
+        'kurtosis':    lambda x: kurtosis(x, axis=1),
+        'line_noise':  lambda x: _freqs_power(x, epochs.info['sfreq'],
+                                              [50, 60]),
+    }
+
+    if picks is None:
+        picks = pick_types(epochs.info, meg=False, eeg=True, exclude=[])
+    if use_metrics is None:
+        use_metrics = metrics.keys()
+
+    # Concatenate epochs in time
+    data = epochs.get_data()
+    data = data.transpose(1, 0, 2).reshape(data.shape[1], -1)
+    data = data[picks]
+
+    # Find bad channels
+    bads = []
+    for m in use_metrics:
+        s = metrics[m](data)
+        b = [epochs.ch_names[picks[i]] for i in find_outliers(s, thres)]
+        logger.info('Bad by %s:\n\t%s' % (m, b))
+        bads.append(b)
+
+    return np.unique(np.concatenate(bads)).tolist()
+
+
+def faster_bad_epochs(epochs, picks=None, thres=3, use_metrics=None):
+    """Implements the second step of the FASTER algorithm.
+    
+    This function attempts to automatically mark bad epochs by performing
+    outlier detection.
+
+    Parameters
+    ----------
+    epochs : Instance of Epochs
+        The epochs to analyze.
+    picks : list of int | None
+        Channels to operate on. Defaults to EEG channels.
+    thres : float
+        The threshold value, in standard deviations, to apply. An epoch
+        crossing this threshold value is marked as bad. Defaults to 3.
+    use_metrics : list of str
+        List of metrics to use. Can be any combination of:
+            'amplitude', 'variance', 'deviation'
+        Defaults to all of them.
+
+    Returns
+    -------
+    bads : list of int
+        The indices of the bad epochs.
+    """
+
+    metrics = {
+        'amplitude': lambda x: np.mean(np.ptp(x, axis=2), axis=1),
+        'deviation': lambda x: np.mean(_deviation(x), axis=1),
+        'variance':  lambda x: np.mean(np.var(x, axis=2), axis=1),
+    }
+
+    if picks is None:
+        picks = pick_types(epochs.info, meg=False, eeg=True, exclude='bads')
+    if use_metrics is None:
+        use_metrics = metrics.keys()
+
+    data = epochs.get_data()[:, picks, :]
+
+    bads = []
+    for m in use_metrics:
+        s = metrics[m](data)
+        b = find_outliers(s, thres)
```

Comment:
for the real re-implementation of faster you need to use the max_iter parameter with 1 here and we should expose it to allow for different values